x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/semantic_search.yml

```diff
@@ -0,0 +1,59 @@
+setup:
+  - skip:
+      features: headers
+  - do:
+      headers:
+        Authorization: "Basic eF9wYWNrX3Jlc3RfdXNlcjp4LXBhY2stdGVzdC1wYXNzd29yZA==" # run as x_pack_rest_user, i.e. the test setup superuser
+      indices.create:
+        index: test
+        body:
+          settings:
+            number_of_replicas: 0
+          mappings:
+            properties:
+              name:
+                type: keyword
+              text_embedding:
+                type: dense_vector
+                dims: 512
+                index: true
+                similarity: cosine
+
+# It is difficult to deploy a text embedding model in yaml tests
+# so focus on negative assertions where the request is invalid
+---
+"unknown deployment":
+  - do:
+      catch: missing
+      semantic_search:
+        index: test
+        body:
+          deployment_id: not-a-model
+          query_string: "kids pyjamas with picture of a tractor"
+          knn:
+            field: vector
+            k: 2
+            num_candidates: 3
```

Comment:
We have an example of doing this in some other yaml tests. Its not too bad. It would be good to fully test it.